src/osm/GeometryHandler.cpp

```diff
@@ -100,2766 +94,187 @@ GeometryHandler<W>::~GeometryHandler() = default;
 // ____________________________________________________________________________
 template <typename W>
 void GeometryHandler<W>::relation(const Relation& rel) {
-  const auto& tags = rel.tags();
-  auto typeTag = tags.find("type");
-  if (typeTag != tags.end() &&
-      (typeTag->second == "boundary" || typeTag->second == "multipolygon")) {
-    for (const auto& member : rel.members()) {
-      if (member.type() == RelationMemberType::WAY &&
-          (member.role() == "outer" || member.role() == "inner")) {
-#pragma omp critical
-        {
-          _areaBorderWaysIndex[member.id()].push_back(
-              {rel.id(), member.role() == "inner"});
-          std::sort(_areaBorderWaysIndex[member.id()].begin(),
-                    _areaBorderWaysIndex[member.id()].end(), MemberRelCmp());
-        }
-      }
-    }
-  }
-}
-
-// ____________________________________________________________________________
-template <typename W>
-void GeometryHandler<W>::area(const Area& area) {
-  // inner simplified geom, empty by default
-  osm2rdf::geometry::Area innerGeom;
-
-  // outer simplified geom, empty by default
-  osm2rdf::geometry::Area outerGeom;
+  if (rel.isArea()) return;  // skip area relations, will be handled by area()
 
-  // the convex hull of the geometry
-  osm2rdf::geometry::Polygon convexHull;
+  if (!rel.hasGeometry()) return;
 
-  const auto& geom = simplifyGeometry(area.geom());
+  const std::string id = _writer->generateIRI(
+      osm2rdf::ttl::constants::RELATION_NAMESPACE[_config.sourceDataset],
+      rel.id());
 
-  size_t totPoints = 0;
-  const size_t MIN_CUTOUT_POINTS = 10000;
+  size_t subId = 0;
 
-  for (const auto& poly : geom) {
-    totPoints += poly.outer().size();
-  }
-
-  if (!_config.dontUseInnerOuterGeoms && (area.hasName() || !area.fromWay())) {
-    innerGeom = simplifiedArea(geom, true);
-    outerGeom = simplifiedArea(geom, false);
-  }
+  if (rel.geom().size() > 1) subId = 1;
 
-  if (!geom.empty()) {
-    convexHull = area.convexHull();
-    if (convexHull.outer().size() >= geom.front().outer().size() / 2) {
-      // if we haven't gained anything significant in terms of number of anchor
-      // points, discard
-      convexHull = osm2rdf::geometry::Polygon();
+  for (const auto& v : rel.geom()) {
+    if (v.getType() == 0) {
```

Comment:
Constants or Enum for `getType()` results?

Reply:
I think this is more of an libutil issue, and should be changed there.